filament/backend/src/metal/MetalContext.h

```diff
@@ -111,6 +117,13 @@ struct MetalContext {
     tsl::robin_set<MetalSamplerGroup*> samplerGroups;
     tsl::robin_set<MetalTexture*> textures;
 
+    // This deque implements delayed destruction for Metal texture handles. It keeps a handle to
+    // the most recent kMetalFreedTextureListSize texture handles. When we're asked to destroy a
+    // texture handle, we free its texture memory, but keep the MetalTexture object alive, marking
+    // it as "terminated". If we later are asked to use that texture, we can check its terminated
+    // status and throw an error instead of crashing.
+    std::deque<Handle<HwTexture>> texturesToDestroy;
```

Comment:
`deque` is very inefficient in terms of code size and heap allocations. I would prefer we use a bespoke, fixed-size, circular buffer. Especially for textures.